packages/material-ui/src/internal/SwitchBase.js

```diff
@@ -30,128 +29,122 @@ export const styles = {
 /**
  * @ignore - internal component.
  */
-class SwitchBase extends React.Component {
-  constructor(props) {
-    super();
-    this.isControlled = props.checked != null;
-    this.state = {};
-    if (!this.isControlled) {
-      // not controlled, use internal state
-      this.state.checked = props.defaultChecked !== undefined ? props.defaultChecked : false;
+const SwitchBase = React.forwardRef(function SwitchBase(props, ref) {
+  const {
+    autoFocus,
+    checked: checkedProp,
+    checkedIcon,
+    classes,
+    className: classNameProp,
+    defaultChecked,
+    disabled: disabledProp,
+    icon,
+    id,
+    inputProps,
+    inputRef,
+    muiFormControl,
+    name,
+    onBlur,
+    onChange,
+    onFocus,
+    readOnly,
+    required,
+    tabIndex,
+    type,
+    value,
+    ...other
+  } = props;
+  const { current: isControlled } = React.useRef(checkedProp != null);
+  const [checkedState, setCheckedState] = React.useState(() => {
+    if (!isControlled) {
+      return defaultChecked !== undefined ? defaultChecked : false;
```

Comment:
This logic was introduced in https://github.com/mui-org/material-ui/commit/5c888657151a81e7a0e1a0de18531eea45d77e5e#diff-9957b5b291ce926258db1099b3707cc2R75. 
What do you think of being consistent with ExpansionPanel.js:
```suggestion
const [checkedState, setCheckedState] = React.useState(defaultChecked);
```

Reply:
I was about to say that. Since we won't read from the state anyway if we're controlled there's not much gained from being smart about the initial state, no? The only thing I would change I would make is add a boolean cast to `defaultChecked` so that we don't get accidental false positives about controlled/uncontrolled switches.